### DISCUSSION TITLE
error：both MaximumAttempts and ExpirationIntervalInSeconds on retry policy are not set, at least one of them must be set

### DISCUSSION THREAD

**Discussion by Johnite** [uber/cadence#4367] [Q&A] [answered]
I created a workflow that consists of three stages, but, only two activitys, the first of which is SQL statement parsing, the second is an activity, do SQL query, and the third stage is another activity, do json coding,but I got this error.  Can you help me?

**Comment by zhangkyou** [comment, marked as answer]
The answer is obviously in the error report. You must set  either MaximumAttempts or ExpirationIntervalInSeconds parameters in ActivityOptions's RetryPolicy config.That two parameters decide your activity execute retry end with MaximumAttempts times or timeout with ExpirationIntervalInSeconds.

**Reply by longquanzheng** [reply]
Some tips on using retry policy and timeout https://stackoverflow.com/questions/65139178/how-to-set-proper-timeout-values-for-cadence-activitieslocal-and-regular-activi/65139179#65139179
